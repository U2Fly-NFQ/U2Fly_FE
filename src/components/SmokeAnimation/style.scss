.ag-smoke-block-container {
  position: relative;
}

.ag-smoke-block {
  position: relative;
  top: 0;
  left: 0;
  height: 200px;
  width: 100%;

  animation: goLeft 4s forwards;
  animation-delay: 1s;
}

.ag-smoke_img {
  width: 100%;
  height: auto;
  object-fit: contain;
  position: absolute;
  bottom: 0;
}

@keyframes goLeft {
  0% {
    left: 1000px;
  }
  50% {
    left: 0;
  }
  100% {
    left: 1000px;
  }
}
