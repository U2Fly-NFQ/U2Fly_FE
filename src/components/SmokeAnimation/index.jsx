import React from 'react'
import './style.scss'

const SmokeAnimation = () => {
  return (
    <>
      <div className="ag-smoke-block-container">
        <div className="ag-smoke-block">
          <picture>
            <source
              srcSet="https://raw.githubusercontent.com/SochavaAG/example-mycode/master/pens/animation-smoke-images/images/smoke-blue.webp"
              type="image/webp"
            />
            <img
              src="https://raw.githubusercontent.com/SochavaAG/example-mycode/master/pens/animation-smoke-images/images/smoke-blue.png"
              alt="smoke-blue"
              className="ag-smoke_img"
              width="1920"
              height="1080"
            />
          </picture>
        </div>
      </div>
    </>
  )
}

export default SmokeAnimation
