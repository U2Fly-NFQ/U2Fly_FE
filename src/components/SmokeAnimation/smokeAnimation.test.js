import renderer from 'react-test-renderer'
import React from 'react'
import SmokeAnimation from './'

describe('Smoke animation test', () => {
  it('renders correctly', () => {
    const tree = renderer.create(<SmokeAnimation />).toJSON()
    expect(tree).toMatchSnapshot()
  })
})
