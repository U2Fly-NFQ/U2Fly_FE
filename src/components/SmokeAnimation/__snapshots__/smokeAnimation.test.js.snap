// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Smoke animation test renders correctly 1`] = `
<div
  className="ag-smoke-block-container"
>
  <div
    className="ag-smoke-block"
  >
    <picture>
      <source
        srcSet="https://raw.githubusercontent.com/SochavaAG/example-mycode/master/pens/animation-smoke-images/images/smoke-blue.webp"
        type="image/webp"
      />
      <img
        alt="smoke-blue"
        className="ag-smoke_img"
        height="1080"
        src="https://raw.githubusercontent.com/SochavaAG/example-mycode/master/pens/animation-smoke-images/images/smoke-blue.png"
        width="1920"
      />
    </picture>
  </div>
</div>
`;
